.github/workflows/build-docker-image.yml

```diff
@@ -15,9 +15,6 @@ jobs:
         go-version: '1.18'
     - run: go version
 
-    - name: Verify project
```

Comment:
I understand that having multiple, smaller GH workflows is probably more maintainable than having a maybe longer pipeline.

But I do believe that we should not build any artifact if the tests fail, that's why I would keep this pipeline as is, or at least make them dependent, with this condition: package the artifacts (docker build) if and only if the tests pass.

Reply:
If it's okay to make it dependent I would prefer that as it's more visible in the checks and runs without the 30m build time. Let me know what you'd prefer

Reply:
I agree with @mdelapenya. It can be just another job in the same workflow

```yml
jobs:
  build:
  build-image:
```

and it will be visible in the checks